### PR TITLE
Fake Data Generator

### PR DESCRIPTION
**Pull request by Jen-Hall**
# Summary

**What/Why**

This pr addresses issue: #3.

This pr includes:
- Documentation on how to generate required json files for forecast model: `hourly_elective_prob.json` and `specialty_info.json`.
- A new data prep module in src to create the json files required for the forecast model.
- A fake data generator which produces the four files needed to run the models: `patient_df.csv`, `historic_admissions.csv`, `hourly_elective_prob.json` and `specialty_info.json`.
- Removes the original json files.
- Converts the csv data dictionaries to json format.
- Update dates in files to align with fake data generator.

**Your approach**
The purpose of `generate_fake_data.py` is to create four files with fake data in the following order: 
- `patient_df.csv`
- `historic_admissions.csv`
- `hourly_elective_prob.json`
- `specialty_info.json`

The fake data is intended for us with the following applications:
- As an example of how data needs to be formatted to be passed into the models.
- To ensure the files are being generated correctly to test GUI setup.
- To test the setup and running of the repo.

The following should be noted with regards to using the fake data generators and the artefacts produced by using fake data:
- *DO NOT* use the model artefacts (`.pkl` files) generated from the fake data to make predictions which will be used in any real world application.
- *DO NOT* use the fake data generated to inform any insights to be applied to a real world setting.
- *DO NOT* use the fake data to test the performance of the model.

The data is generated completely randomly, with each field having random values generated independently of other fields. This generator was created having never been exposed to the real data.

Note a number of the fields will show only test categories and may not be reflective of the types of categories or number of categories types in each field. However the formatting of the fields is reflective as an example of what is required to run the models.

**What to review**
- Ensure the json dictionaries match the csv formatted data dictionaries.
- The fake data files generated from the fake data generation work with models

# Checklists

This pull/merge request meets the following requirements:

- [x] code runs
- [x] [developments are ethical][data-ethics-framework]
- [x] developments are secure (no data or notebook output)
- [x] you have made proportionate checks that the code works correctly
- [x] [minimum usable documentation][agilemodeling] written in the `docs` folder

Comments have been added below around the incomplete checks.

[agilemodeling]: http://agilemodeling.com/essays/documentLate.htm
[data-ethics-framework]: https://www.gov.uk/government/publications/data-ethics-framework

### CODE REVIEW

**Comment by amadeuspzs** [comment]
1. `historic_admissions.csv` needs an entry for every hour, including at least one week until after the "current date" for the forecast to render.
2. `src/forecasting/hourly_elective_prob.json` needs to be generated from the fake data
3. `src/forecasting/specialty_info.json` needs to be generated from the fake data

**Comment by Jen-Hall** [comment]
> 1. `historic_admissions.csv` needs an entry for every hour, including at least one week until after the "current date" for the forecast to render.
> 2. `src/forecasting/hourly_elective_prob.json` needs to be generated from the fake data
> 3. `src/forecasting/specialty_info.json` needs to be generated from the fake data

Thank you I will also update the `README.md` and data dictionaries.

**Comment by amadeuspzs** [comment]
@Jen-Hall  could you:

1. Rename df_prob.json -> hourly_elective_prob.json in the generator
2. Typo in README: "This directory contains two files called" -> one file
3. Can you include the date range used in the README/usage, and update https://github.com/nhsx/skunkworks-bed-allocation/blob/143aa014ca6e852bde9b516194f342f3ac9288ac/src/forecasting/utils.py#L9 to match a date that is at least one week before the end of the date range used?
4. Can you remove the previously committed .json files

File: fake_data_generation/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+└───fake_data_files
+    │   patient_df.csv
+    │   historic_admissions.csv
+    |   hourly_elective_prob.json
+    |   specialty_info.json
+```
+
+
+### How to run
+Before running ensure your environment is set up as described in: [README.md](../README.md).
```
Comment (amadeuspzs) [inline fake_data_generation/README.md:51]: Can you clarify this needs to be in "development" mode (https://github.com/nhsx/skunkworks-bed-allocation/blob/2fd925ff14fe207089f9838a86ea501c029b9df0/README.md?plain=1#L63)

**Review by amadeuspzs — CHANGES_REQUESTED** [review]

**Review by amadeuspzs — APPROVED** [review]

**Comment by amadeuspzs** [comment]
I can confirm integration test runs successfully: 
image

**Review by matthewcooper19 — APPROVED** [review]
No changes from me

**Pull request merged**